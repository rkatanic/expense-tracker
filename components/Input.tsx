interface Props {
  placeholder?: string;
  value?: string | number;
  type?: "text" | "number" | "email" | "password" | "date" | "radio";
  id?: string;
  name?: string;
  onChange?: (event?: any) => void;
  checked?: boolean;
  label?: string;
  pattern?: string;
  min?: number;
  max?: number;
  required?: boolean;
  size?: "large" | "medium";
}

const Input = ({
  placeholder,
  value,
  type = "text",
  id,
  name,
  onChange,
  checked,
  label,
  pattern,
  min,
  max,
  required,
  size,
}: Props): JSX.Element => {
  const isRadio = type === "radio";

  return (
    <div className="w-full flex flex-col">
      {label && (
        <label
          className="block mb-1 text-sm font-semibold text-zinc-900 dark:text-zinc-400"
          htmlFor=""
        >
          {label}
        </label>
      )}
      <input
        className={`${
          size === "large" ? "py-2" : "py-1.5"
        } px-3 w-full shadow-sm text-sm text-zinc-900 rounded-md border bg-white border-zinc-200  focus:outline-emerald-500 dark:focus:outline-0 dark:focus:border-emerald-500 dark:border-zinc-700 dark:text-zinc-100 dark:bg-zinc-800/70`}
        required={required}
        min={min}
        max={max}
        pattern={pattern}
        type={type}
        value={value}
        placeholder={placeholder}
        id={id}
        name={name}
        onChange={onChange}
        checked={checked}
      />
    </div>
  );
};

export default Input;
