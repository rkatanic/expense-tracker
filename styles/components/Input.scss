.input {
  font-size: 0.875rem;
  background-color: transparent;
  border: none;
  border-bottom: 2px solid var(--color-gray-600);
  padding: 0.5rem 0;
  outline: none;
  color: var(--color-white-one)
}

.input:focus {
  transition: 0.5s;
  border-bottom-color: var(--color-gray-800);
}

.input::placeholder {
  color: var(--color-gray-700);
}