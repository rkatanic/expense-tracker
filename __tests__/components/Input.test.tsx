import { fireEvent, render, waitFor } from "@testing-library/react";
import Input from "../../components/Input";

describe("Input", (): void => {
  it("should render", () => {
    const { baseElement } = render(
      <Input placeholder="Input" value="value" onChange={jest.fn()} />
    );

    expect(baseElement).toMatchSnapshot();
  });

  it("should call onChange handler", (): void => {
    const mockOnChange = jest.fn();
    const { getByText, getByPlaceholderText } = render(
      <Input placeholder="Input" value="value" onChange={mockOnChange} />
    );

    fireEvent.change(getByPlaceholderText("Input"), {
      target: { value: "new value" },
    });

    expect(mockOnChange).toHaveBeenCalledTimes(1);
    waitFor(() => {
      expect(getByText("new value")).toBeInTheDocument();
    });
  });
});
