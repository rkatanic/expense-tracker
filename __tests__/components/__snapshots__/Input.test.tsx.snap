// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render 1`] = `
<body>
  <div>
    <div
      class="w-full flex flex-col"
    >
      <input
        class="py-1.5 px-3 w-full text-sm text-zinc-900 rounded-md border bg-zinc-50 border-zinc-300 focus:border-sky-400 focus:ring-sky-400 dark:focus:ring-sky-500 focus:outline-0 dark:border-zinc-500 dark:text-zinc-100 dark:bg-zinc-900"
        placeholder="Input"
        type="text"
        value="value"
      />
    </div>
  </div>
</body>
`;
